src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -328,6 +340,8 @@ public function request($method, $uri, array $parameters = array(), array $files
         }
 
         if ($this->followRedirects && $this->redirect) {
+            $this->redirects[] = spl_object_hash($this->history->current());
```

Comment:
you should use the hash as key rather than value. the array is used to check whether it contains a given hash, and `O(1)` is better than than `O(n)`